### ISSUE TITLE
Can't draw if dragging finger horizontally

### ISSUE THREAD

**Issue by Odaym** [PaNaVTEC/DrawableView#3]
Tested this on Galaxy S3 and Nexus 5. I cannot seem to be able to draw if I start dragging my finger horizontally when trying to draw anything, the line barely starts and then it cuts off every time. I can draw just fine if I drag diagonally or vertically and that's perfectly fine.

Configuration while facing this (I have 1 and 1 set for Min/Max zoom because I don't want to enable any zoom ability):

```java
config.setStrokeColor(getResources().getColor(android.R.color.white));
            config.setStrokeWidth(20.0f);
            config.setMinZoom(1.0f);
            config.setMaxZoom(1.0f);
            config.setCanvasWidth(4000);
            config.setCanvasHeight(4000);
```

**Comment by PaNaVTEC** [comment]
I'm trying to reproduce, but I can't do it. I'm trying with a Nexus 7, Nexus 6 and an LG G2. I setted the sample to use this configuration. Do you know that dragging with 2 fingers is to move the canvas and dragging with one finger is for painting?  I'm trying to play with both configurations and I can't reproduce. Can you show me the Layout XML/activity what are you using?  I'm really interested in catch this bug.

**Comment by Odaym** [comment]
Here you will find a small video showing the bug: https://dl.dropboxusercontent.com/u/19390574/drawableView_horizontal.mp4

And here's my XML of that view, this is all a fragment that's sitting in a view pager. The config is set to the drawableView inside the onCreateView() method of the Fragment

```xml
<RelativeLayout xmlns:android="http://schemas.android.com/apk/res/android"
    android:layout_width="match_parent"
    android:layout_height="match_parent">

    <ImageView
        android:id="@+id/bookmarkIMG"
        android:layout_width="match_parent"
        android:layout_height="wrap_content"
        android:layout_centerInParent="true" />

    <me.panavtec.drawableview.DrawableView
        android:id="@+id/drawable_view"
        android:layout_width="wrap_content"
        android:layout_height="wrap_content"/>

    <ProgressBar/>

    <RelativeLayout>

        <TextView/>

        <TextView/>

        <TextView />

        <TextView/>

    </RelativeLayout>

    <ImageButton/>

</RelativeLayout>
```

**Comment by PaNaVTEC** [comment]
Ah man! That's the problem you are using a ViewPager so your View Hierarchy is consuming the horizontal event before the view catches it. To solve this, use a custom wrapper for your DrawableView, you can face the same problem if you use a HorizontalScrollView inside a ViewPager or some other component that uses horizontal touches. Here is a solution, make a new class that extends from the DrawableView and override the "dispatchTouchEvent" in this way:

```java
@Override public boolean dispatchTouchEvent(MotionEvent event) {
    if (event.getAction() == MotionEvent.ACTION_MOVE && getParent() != null) {
      getParent().requestDisallowInterceptTouchEvent(true);
    }
    return super.dispatchTouchEvent(event);
  }
```

And in your fragment XML use your new class. Tell me if it works.

**Comment by Odaym** [comment]
Makes sense, that's what I noticed too when I just finished recording the video but didn't act upon it. To avoid UX confusion Im not going to try and solve this because this is not the only problem that will arise from having both enabled at the same time (scrolling and drawing). I will make the drawable view in a non-moving way over the view pager and then the view pager can be returned to when the drawing is over, much clearer way.

Thanks for helping so far though :)
